src/tests/ffi-key-sig.cpp

```diff
@@ -508,20 +533,46 @@ TEST_F(rnp_tests, test_ffi_export_revocation)
 
     /* check signature contents */
     pgp_source_t src = {};
-    assert_rnp_success(init_file_src(&src, "alice-revocation.pgp"));
+    pgp_source_t file_src = {};
+    pgp_source_t armored_src = {};
+    assert_rnp_success(init_file_src(&file_src, "alice-revocation.pgp"));
```

Comment:
Same here: better duplicate tests code then make it complicated to debug afterwards.
Also, as these are FFI tests, we should not use internal functions from the lower layer (pgp_source_t and so on).
Reason for this is that at some point it would be desirable to run FFI tests suite against the built library instead of compiling tests together with lower layer sources.

We have helper functions `file_to_str()`/`file_to_vec()` in the `support.h`.

Reply:
> I've disambiguated the scenarios.

Please see my previous comment - we do not need to run all bunch of tests for armored/binary output.

> I see, it's testing that the revocation certificate meets the expectations, but does so using internal API.

Yeah, you are right (missed that we had lowerl-layer dependency already) - let's leave it as is and add a ticket for that. It should be easier to do this check via CLI tests, by exporting revocation and running `rnpkeys --list-packets` on it.